shadowsocks-csharp/View/HotkeySetting.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Shadowsocks.Controller;
+using Shadowsocks.Model;
+using Shadowsocks.Properties;
+using System;
+using System.Drawing;
+using System.Text;
+using System.Windows.Forms;
+
+namespace Shadowsocks.View {
+    public partial class HotkeySetting : Form {
+        ShadowsocksController controller;
+
+        public HotkeySetting(ShadowsocksController controller) {
+            this.controller = controller;
+            InitializeComponent();
+            Icon = Icon.FromHandle(Resources.ssw128.GetHicon());
+        }
+
+        /// <summary>
+        /// 初始化加载窗体
+        /// </summary>
+        private void HotkeySetting_Load(object sender, EventArgs e) {
+            I18n();
+
+            //Load Config
+            HotkeyConfig conf = controller.GetConfigurationCopy().hotkey;
+            if(conf == null) {
+                conf = new HotkeyConfig();
+            }
+            textBox1.Text = conf.SwitchSystemProxy;
+            textBox2.Text = conf.ChangeToPac;
+            textBox3.Text = conf.ChangeToGlobal;
+            textBox4.Text = conf.SwitchAllowLan;
+            textBox5.Text = conf.ShowLogs;
+            checkBox1.Checked = conf.AllowSwitchServer;
+        }
+
+        /// <summary>
+        /// 窗口文本Internationalization
+        /// </summary>
+        private void I18n() {
+            Text = I18N.GetString(Text);
+            label1.Text = I18N.GetString(label1.Text);
+            label2.Text = I18N.GetString(label2.Text);
+            label3.Text = I18N.GetString(label3.Text);
+            label4.Text = I18N.GetString(label4.Text);
+            label5.Text = I18N.GetString(label5.Text);
+            checkBox1.Text = I18N.GetString(checkBox1.Text);
+            ok.Text = I18N.GetString(ok.Text);
+            cancel.Text = I18N.GetString(cancel.Text);
+        }
+
+        private StringBuilder sb = new StringBuilder();
+
+        /// <summary>
+        /// 快捷键捕获 - 按下键盘
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e">键</param>
+        private void HotkeyDown(object sender, KeyEventArgs e) {
+            sb.Length = 0;
+            //仅允许组合键
+            if(e.Modifiers != 0) {
+                if(e.Control) {
+                    sb.Append("Ctrl + ");
+                }
+                if(e.Alt) {
+                    sb.Append("Alt + ");
+                }
+                if(e.Shift) {
+                    sb.Append("Shift + ");
+                }
+                if((e.KeyValue >= 33 && e.KeyValue <= 40) ||
+                    (e.KeyValue >= 65 && e.KeyValue <= 90) ||
+                    (e.KeyValue >= 112 && e.KeyValue <= 123)) {
+                    sb.Append(e.KeyCode);
+                } else if(e.KeyValue >= 48 && e.KeyValue <= 57) {
+                    sb.Append('D').Append((char)e.KeyValue);
+                } else if(e.KeyValue >= 96 && e.KeyValue <= 105) {
```

Comment:
It mean the key code.
33~40 is the function keys on keyboard.
65~90 means A to Z on keyboard.
112~123 means F1 to F12 on keyboard.
48~57 means 0 to 9 number keys above letter keys on keyboard.
96~105 means 0 to 9 number keys at right of the keyboard.
You can google "Keycode" to get more information.